### PR TITLE
Parser Options

### PR DESCRIPTION
**Pull request by djbe**
Makes the necessary config/cli changes to pass along an `options` dictionary to the parser.

Note: 1st step for #576 (and others).

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Sources/SwiftGen/Commander/ParserCLICommands.swift [hunk 1]
```diff
@@ -61,13 +67,15 @@ extension ParserCLI {
...
       CLIOption.params,
       CLIOption.filter(for: self),
       OutputDestination.cliOption,
       VariadicArgument<Path>("PATH", description: self.pathDescription, validator: pathsExist)
-    ) { oldTemplateName, templateName, templatePath, parameters, filter, output, paths in
+    ) { oldTemplateName, templateName, templatePath, parserOptions, parameters, filter, output, paths in
       try ErrorPrettifier.execute {
-        let parser = try self.parserType.init(options: [:]) { msg, _, _ in
+        let options = try Parameters.parse(items: parserOptions)
+        let parser = try self.parserType.init(options: options) { msg, _, _ in
```
Comment (AliSoftware) [inline Sources/SwiftGen/Commander/ParserCLICommands.swift:78]: Gotta love it when everything comes together 👌

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,7 +10,9 @@ _None_
 
 ### New Features
 
-_None_
+* Command now accept an `options` dictionary, with which you can set internal parser settings to change its behaviour. See the commands' specific documentation for available options.
```
Comment (AliSoftware) [inline CHANGELOG.md]: "Each parser command now accepts…"

File: Documentation/Commands/colors.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+You can provide a custom filter using the `filter` option, it accepts any valid regular expression. See the [Config file documentation](../ConfigFile.md) for more information.
+
+## Customization
+
+This parser currently doesn't accept any options.
+
+## Templates
+
+[See here](../templates/colors) for a list of templates and their documentation. If you want to write custom templates, make sure to check the [stencil context documentation](../SwiftGenKit%20Contexts/Colors.md) to see what data is available after parsing.
```
Comment (AliSoftware) [inline Documentation/Commands/colors.md]: Maybe split that in two bullet points.

Also `s/for a list of templates/for a list of bundled templates/` or `for a list of templates bundled with SwiftGen`?

File: Documentation/Commands/coredata.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+You can provide a custom filter using the `filter` option, it accepts any valid regular expression. See the [Config file documentation](../ConfigFile.md) for more information.
+
+## Customization
+
+This parser currently doesn't accept any options.
+
+## Templates
+
+[See here](../templates/coredata) for a list of templates and their documentation. If you want to write custom templates, make sure to check the [stencil context documentation](../SwiftGenKit%20Contexts/CoreData.md) to see what data is available after parsing.
```
Comment (AliSoftware) [inline Documentation/Commands/coredata.md]: Same here and obviously for the others too

File: Documentation/Commands/fonts.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# Fonts parser
+
+## Input
+
+The fonts parser accepts a directory, which it'll recursively search for font files. Supported file types depend on the target platform, for example iOS only supports TTF and OTF font files.
```
Comment (AliSoftware) [inline Documentation/Commands/fonts.md:5]: Is that a fact?

File: Documentation/Commands/ib.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Interface Builder parser
+
+## Input
+
+The interface builder parser accepts either a file or a directory, which it'll search for `storyboard` files. The parser will load all the scenes and all the segues for each storyboard. It supports both AppKit (macOS) and UIKit (iOS, watchOS, tvOS) storyboards.
+
+### Filter
+
+The default filter for this command is: `[^/]\.storyboard$`. That means it'll accept any file with the extension `storyboard`.
```
Comment (AliSoftware) [inline Documentation/Commands/ib.md:9]: At some point we discussed about keeping the (now deprecated) `storyboards` subcommand and give it a default filter of `.storyboard$` while the `ib` subcommand was gonna default to `.(xib|storyboard)$`… do we still want to do that? Or will we just switch the default filter for `ib` when we start spring XIBs (for reuseIdentifiers and stuff) and continue deprecating the storyboard command?

File: Documentation/Commands/strings.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+### Filter
+
+The default filter for this command is: `[^/]\.strings$`. That means it'll accept any file with the extension `strings`.
+
+You can provide a custom filter using the `filter` option, it accepts any valid regular expression. See the [Config file documentation](../ConfigFile.md) for more information.
+
+## Customization
+
+This parser currently doesn't accept any options.
```
Comment (AliSoftware) [inline Documentation/Commands/strings.md:39]: 🔜😜 (Next PR I guess)

File: Documentation/Commands/xcassets.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
+# Asset Catalog parser
+
+## Input
+
+The assets parser accepts one (or more) asset catalogs, which it'll parse for supported set types and groups. We currently support the following types:
+- Group Type
```
Comment (AliSoftware) [inline Documentation/Commands/xcassets.md]: That's what's usually called a "folder" in the asset catalog, right? Might want to mention that in parentheses maybe

File: Documentation/ConfigFile.md [hunk 9]
```diff
@@ -62,6 +62,7 @@ Each key corresponding to a SwiftGen subcommands (`colors`, `coredata`, `fonts`,
 | Subkey | Type | Description |
 |--------|------|-------------|
 | `inputs` | Path or Array of Paths | The file(s)/dir(s) to parse (e.g. the path to your assets catalog for the `xcassets` command, or your `Localizable.strings` file for the `strings` command, etc). |
+| `options` | Dictionary | Any optional list of settings for the parser, to change it's behaviour. See the commands' specific documentation for available options. |
```
Comment (AliSoftware) [inline Documentation/ConfigFile.md]: `s/it's/its/`

**Review by AliSoftware — COMMENTED** [review]
That was fast 👍

File: Documentation/Commands/fonts.md (hunk 5 above)
Comment (djbe) [reply Documentation/Commands/fonts.md:5]: From some Apple documentation file:
> Each payload must contain exactly one font file in TrueType (.ttf) or OpenType (.otf) format. Collection formats (.ttc or .otc) are not supported.

File: Documentation/Commands/ib.md (hunk 6 above)
Comment (djbe) [reply Documentation/Commands/ib.md:9]: Depends on when we get `xib` support in. If it's not ready by 7.0, we may as well drop the alias 🤷‍♂️

File: Documentation/Commands/strings.md (hunk 7 above)
Comment (djbe) [reply Documentation/Commands/strings.md:39]: 😆

**Comment by djbe** [comment]
Yeah I started writing it when I saw your PR (on top of yours), it was ready when yours finished review. 😄

File: Documentation/Commands/ib.md (hunk 6 above)
Comment (AliSoftware) [reply Documentation/Commands/ib.md:9]: Yeah, and given in the future there might be many applications, like reuseIdentifiers, accessibilityIndentifiers, etc… not sure the filter would be worth it. We'll see when we start tackling it anyway.

**Review by AliSoftware — APPROVED** [review]

**Pull request merged**